x-pack/plugins/security_solution/public/common/components/events_viewer/index.tsx

```diff
@@ -170,78 +215,402 @@ const StatefulEventsViewerComponent: React.FC<Props> = ({
     ) : null;
   }, [graphEventId, tableId, sessionViewConfig, SessionView, Navigation]);
   const setQuery = useCallback(
-    (inspect, loading, refetch) => {
+    ({ id, inspect, loading, refetch }: SetQuery) =>
       dispatch(
         inputsActions.setQuery({
-          id: tableId,
+          id,
           inputId: InputsModelId.global,
           inspect,
           loading,
           refetch,
         })
-      );
-    },
-    [dispatch, tableId]
+      ),
+    [dispatch]
   );
 
   const fieldBrowserOptions = useFieldBrowserOptions({
-    sourcererScope: scopeId,
+    sourcererScope,
     editorActionsRef,
     upsertColumn: (column, index) =>
       dispatch(dataTableActions.upsertColumn({ column, id: tableId, index })),
     removeColumn: (columnId) => dispatch(dataTableActions.removeColumn({ columnId, id: tableId })),
   });
 
-  const isLive = input.policy.kind === 'interval';
+  const columnHeaders = isEmpty(columns) ? defaultHeaders : columns;
+  const esQueryConfig = getEsQueryConfig(uiSettings);
+
+  const filterQuery = useMemo(
+    () =>
+      getCombinedFilterQuery({
+        config: esQueryConfig,
+        browserFields,
+        dataProviders: [],
+        filters: globalFilters,
+        from: start,
+        indexPattern,
+        kqlMode: 'filter',
+        kqlQuery: query,
+        to: end,
+      }),
+    [esQueryConfig, browserFields, globalFilters, start, indexPattern, query, end]
+  );
+
+  const canQueryTimeline = useMemo(
+    () =>
+      filterQuery != null &&
+      isLoadingIndexPattern != null &&
+      !isLoadingIndexPattern &&
+      !isEmpty(start) &&
+      !isEmpty(end),
+    [isLoadingIndexPattern, filterQuery, start, end]
+  );
+
+  const fields = useMemo(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    () => [...columnHeaders.map((c: { id: any }) => c.id), ...(queryFields ?? [])],
```

Comment:
Can the ID be anything besides `string` or maybe `number`? If we can be explicit here, that'd be great otherwise, maybe `unknown` is more accurate here